src/authInfo.ts

```diff
@@ -972,7 +916,10 @@ export class AuthInfo extends AsyncCreatable<AuthInfo.Options> {
   // Build OAuth config for a JWT auth flow
   private async buildJwtConfig(options: OAuth2Options): Promise<AuthFields> {
     const privateKeyContents = await fs.readFile(ensure(options.privateKey), 'utf8');
-    const audienceUrl = await getJwtAudienceUrl(options);
+    const { loginUrl = '' } = options;
```

Comment:
Why make an empty string the default?

Won't that throw an error like this one if `loginUrl` isn't defined in `options`?
![image](https://user-images.githubusercontent.com/1084688/122769834-eaa94a80-d272-11eb-8735-34b34cbebdd4.png)

And if so, could we at least show a slightly better error message like "loginUrl is required"?

Reply:
This is a good catch, the original version of `getJwtAudienceUrl` would return `SfdcUrl.PRODUCTION` if no `loginUrl` is given, so we have that option, or we can throw with a good message, what would be our best option ?

Reply:
Ah, I'll default to someone with more historical context than I here.
@amphro or @mdonnalley thoughts?

I personally would be fine with returning `SfdxUrl.PRODUCTION` though.